### ISSUE TITLE
NetMHCpan: False peptide min/max boundaries

### ISSUE THREAD

**Issue by jonasscheid** [KohlbacherLab/epytope#91]
The internal boundaries of netmhcpan regarding peptide lengths are 8 - 56. NetMHCpan 4.1 always picks the most relevant Core of the peptide and does prediction on it. That is currently prohibited by the supported length that is inherited by the [netmhcpan 2.8 interface](https://github.com/KohlbacherLab/epytope/blob/28d70d4074194b5e604dd38f0fc973dce2e37ded/epytope/EpitopePrediction/External.py#L648).

Regarding NetMHCIIpan there seems to be only a lower limit to 9mers, I can go beyond 100 amino acids, which clearly does not make sense. So we could also say that we restrict it to peptides having max length of 56

**Comment by jonasscheid** [comment]
I does not lead to wrong results, but class 1 peptides > 14 cannot be predicted by netmhcpan. Also netmhcIIpan is [capped](https://github.com/KohlbacherLab/epytope/blob/28d70d4074194b5e604dd38f0fc973dce2e37ded/epytope/EpitopePrediction/External.py#L1496) to peptides <20, which is a problem if we want to predict 21, 22mers. I think we don't need to change anything in parsing, it's just the boundaries of the peptide lengths.

What I wanted to say is that netmhc gets the most interesting core of the peptide and does the prediction on that:
![Screenshot 2023-07-24 at 17 41 02](https://github.com/KohlbacherLab/epytope/assets/43858870/6f878018-a006-472b-a6cc-b914dfe2f9f5)

**Comment by christopher-mohr** [comment]
I am still not sure I understand how the core is related to the problem with not having allowed all possible peptide lengths. The core is always the 9mer that is predicted to be in contact with the HLA molecule. I think the prediction is still done on the whole peptide though but would have to look up the details.

**Comment by jonasscheid** [comment]
You are right, I checked prediction with the core preptide and the rank is different. So let's allow all possible peptide lengths then?

**Comment by christopher-mohr** [comment]
Sounds good to me, still worth to be changed. Thanks for bringing it up.
